pkg/crypto/cert_accept_reject_test.go

```diff
@@ -0,0 +1,392 @@
+package crypto
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/apiserver/pkg/authentication/user"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// rootAlfa signs signerBravo
+// signerBravo signs signerDelta
+// signerDelta signs servingCert
+func TestServingCertVerification(t *testing.T) {
+	rootAlphaConfig, err := MakeSelfSignedCAConfigForDuration("RootAlfa", 1*time.Hour)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rootAlpha := &CA{
+		Config:          rootAlphaConfig,
+		SerialGenerator: &RandomSerialGenerator{},
+	}
+	rootAlfaPEMBytes, _, err := rootAlphaConfig.GetPEMBytes()
+	if err != nil {
+		t.Fatal(err)
+	}
+	signerBravoConfig, err := MakeCAConfigForDuration("SignerBravo", 1*time.Hour, rootAlpha)
+	if err != nil {
+		t.Fatal(err)
+	}
+	signerBravoConfig.Certs = signerBravoConfig.Certs[:1] // only keep ourselves, not the parents
+	signerBravo := &CA{
+		Config:          signerBravoConfig,
+		SerialGenerator: &RandomSerialGenerator{},
+	}
+	signerBravoPEMCertByte, _, err := signerBravoConfig.GetPEMBytes()
+	if err != nil {
+		t.Fatal(err)
+	}
+	signerDeltaConfig, err := MakeCAConfigForDuration("SignerDelta", 1*time.Hour, signerBravo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	signerDeltaConfig.Certs = signerDeltaConfig.Certs[:1] // only keep ourselves, not the parents
+	signerDelta := &CA{
+		Config:          signerDeltaConfig,
+		SerialGenerator: &RandomSerialGenerator{},
+	}
+	signerDeltaPEMCertByte, _, err := signerDeltaConfig.GetPEMBytes()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	servingCertConfig, err := signerDelta.MakeServerCertForDuration(sets.NewString("::", "127.0.0.1", "localhost"), 1*time.Hour)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// strip the intermediate signers which are included
+	servingCertConfig.Certs = servingCertConfig.Certs[:1]
+	servingPEMCertByte, servingPEMKeyBytes, err := servingCertConfig.GetPEMBytes()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Log(string(servingPEMCertByte))
```

Comment:
We should probably remove this or print other certs out as well. I'm fine with either way.